### PR TITLE
3725 fixes download test

### LINKED ISSUE #3725: nightly test: tests.test_download_and_extract
**Describe the bug**
```
======================================================================
FAIL: test_default (tests.test_download_and_extract.TestDownloadAndExtract)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jenkins/agent/workspace/Monai-pytorch-versions/tests/test_download_and_extract.py", line 65, in test_default
    file_type="tar",
RuntimeError: Download of file from https://drive.google.com/uc?id=1HrQd-AKPbts9jkTNN4pT8vLZyhM5irVn to /tmp/tmp5dkug159/uc?id=1HrQd-AKPbts9jkTNN4pT8vLZyhM5irVn failed due to network issue or denied permission.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/jenkins/agent/workspace/Monai-pytorch-versions/tests/test_download_and_extract.py", line 78, in test_default
    self.assertTrue(str(e).startswith("md5 check"))
AssertionError: False is not true

----------------------------------------------------------------------
```

and additional issues such as:

```

======================================================================
ERROR: test_pretrain_consistency_0 (tests.test_densenet.TestPretrainedDENSENET)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/dist-packages/parameterized/parameterized.py", line 533, in standalone_func
    return func(*(a + p.args), **p.kwargs)
  File "/home/jenkins/agent/workspace/MONAI-postmerge/monai_github/tests/test_densenet.py", line 95, in test_pretrain_consistency
    net = model(**input_param).to(device)
  File "/home/jenkins/agent/workspace/MONAI-postmerge/monai_github/monai/networks/nets/densenet.py", line 311, in __init__
    _load_state_dict(self, "densenet121", progress)
  File "/home/jenkins/agent/workspace/MONAI-postmerge/monai_github/monai/networks/nets/densenet.py", line 276, in _load_state_dict
    state_dict = load_state_dict_from_url(model_url, progress=progress)
  File "/usr/local/lib/python3.9/dist-packages/torch/hub.py", line 586, in load_state_dict_from_url
    download_url_to_file(url, cached_file, hash_prefix, progress=progress)
  File "/usr/local/lib/python3.9/dist-packages/torch/hub.py", line 474, in download_url_to_file
    buffer = u.read(8192)
  File "/usr/lib/python3.9/http/client.py", line 463, in read
    n = self.readinto(b)
  File "/usr/lib/python3.9/http/client.py", line 507, in readinto
    n = self.fp.readinto(b)
  File "/usr/lib/python3.9/socket.py", line 704, in readinto
    return self._sock.recv_into(b)
  File "/usr/lib/python3.9/ssl.py", line 1241, in recv_into
    return self.read(nbytes, buffer)
  File "/usr/lib/python3.9/ssl.py", line 1099, in read
    return self._sslobj.read(len, buffer)
ConnectionResetError: [Errno 104] Connection reset by peer

----------------------------------------------------------------------

```

### PR DESCRIPTION
**Pull request by wyli**
Signed-off-by: Wenqi Li <wenqil@nvidia.com>

Fixes #3725

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: tests/utils.py [hunk 1]
```diff
@@ -93,17 +94,25 @@ def assert_allclose(
...
+def skip_if_downloading_fail():
     try:
+        yield
+    except (ContentTooShortError, HTTPError, ConnectionError) as e:
+        raise unittest.SkipTest(f"error while downloading: {e}") from e
+    except RuntimeError as rt_e:
+        if "network issue" in str(rt_e):
+            raise unittest.SkipTest(f"error while downloading: {rt_e}") from rt_e
+        if "gdown dependency" in str(rt_e):  # no gdown installed
+            raise unittest.SkipTest(f"error while downloading: {rt_e}") from rt_e
```
Comment (Nic-Ma) [inline tests/utils.py:107]: Will this `rt_e` error explicitly say something about installing `gdown`?

Thanks.

**Review by Nic-Ma — APPROVED** [review]
Hi @wyli ,

Is it PR ready for review now? I saw you updated a lot after requesting review.
But it overall looks good to me.
Add @ericspod and @rijobro to double confirm the `contextmanager` as you guys are python feature experts.

Thanks for the enhancement.

File: tests/utils.py (hunk 1 above)
Comment (wyli) [reply tests/utils.py:107]: yes it's from this https://github.com/Project-MONAI/MONAI/blob/b1a96c571fa9ec3dccf7e16a0105df78d6f00889/monai/apps/utils.py#L191

**Pull request merged**